src/clients/MultiCallerClient.ts

```diff
@@ -214,12 +214,35 @@ export class MultiCallerClient {
         batchTxn: { ...transaction, contract: transaction.contract.address },
         reason,
       });
-      batchTxns[idx].gasLimit = succeed ? transaction.gasLimit : undefined;
+      if (transaction.method === "tryMulticall") {
+        const succeededTxnRequests: AugmentedTransaction[] = [];
+        data.forEach(({ success }, subIdx) => {
+          if (success) {
+            const txnIdx =
+              this.chunkSize == undefined
+                ? this.chunkSize[chainId] * idx + subIdx
+                : DEFAULT_MULTICALL_CHUNK_SIZE * idx + subIdx;
```

Comment:
Something like this instead?

```suggestion
            const txnIdx = (this.chunkSize[chainId] ?? DEFAULT_MULTICALL_CHUNK_SIZE) * idx + subIdx;
```

Reply:
Done. https://github.com/across-protocol/relayer-v3/pull/1466/commits/ad5ff97c0c7b5ffa62684f45d0a80f74f3f9766c